engine/api/application_group.go

```diff
@@ -240,3 +245,132 @@ func (api *API) deleteGroupFromApplicationHandler() Handler {
+		}
+
+		f, errF := exportentities.GetFormat(format)
+		if errF != nil {
+			return sdk.WrapError(sdk.ErrWrongRequest, "importGroupsInApplication> Unable to get format")
+		}
+
+		var errorParse error
+		switch f {
+		case exportentities.FormatJSON:
+			errorParse = json.Unmarshal(data, &groupsToAdd)
+		case exportentities.FormatYAML:
+			errorParse = yaml.Unmarshal(data, &groupsToAdd)
+		}
+
+		if errorParse != nil {
+			return sdk.WrapError(sdk.ErrWrongRequest, "importGroupsInApplication> Cannot parsing")
+		}
+
+		groupsToAddInProj := []sdk.GroupPermission{}
+		for _, gr := range groupsToAdd {
+			exist := false
+			for _, gro := range proj.ProjectGroups {
+				if gr.Group.Name == gro.Group.Name {
+					exist = true
+				}
+			}
+			if !exist && !forceUpdate {
+				return sdk.WrapError(sdk.ErrGroupNotFound, "importGroupsInApplication> Group %v doesn't exist in this project", gr.Group.Name)
+			} else if !exist && forceUpdate {
+				groupsToAddInProj = append(groupsToAddInProj, sdk.GroupPermission{
+					Group:      gr.Group,
+					Permission: permission.PermissionRead,
+				})
+			}
+		}
+
+		tx, errBegin := api.mustDB().Begin()
+		if errBegin != nil {
+			return sdk.WrapError(errBegin, "importGroupsInApplication> Cannot start transaction")
+		}
+		defer tx.Rollback()
+
+		if forceUpdate { // clean and update
+			for _, gr := range groupsToAddInProj {
+				gro, errG := group.LoadGroup(tx, gr.Group.Name)
+				if errG != nil {
+					return sdk.WrapError(sdk.ErrGroupNotFound, "importGroupsInApplication> Group %v doesn't exist", gr.Group.Name)
+				}
+				if err := group.InsertGroupInProject(tx, proj.ID, gro.ID, gr.Permission); err != nil {
+					return sdk.WrapError(err, "importGroupsInApplication> Cannot add group %v in project %s", gr.Group.Name, proj.Name)
+				}
+				gr.Group = *gro
+				proj.ProjectGroups = append(proj.ProjectGroups, gr)
+			}
+
+			if err := group.DeleteAllGroupFromApplication(tx, app.ID); err != nil {
+				return sdk.WrapError(err, "importGroupsInApplication> Cannot delete all groups for this application %s", app.Name)
+			}
+
+			for _, gr := range groupsToAdd {
+				gro, errG := group.LoadGroup(tx, gr.Group.Name)
+				if errG != nil {
+					return sdk.WrapError(sdk.ErrGroupNotFound, "importGroupsInApplication> Cannot load group %s : %s", gr.Group.Name, errG)
+				}
+				if err := group.InsertGroupInApplication(tx, app.ID, gro.ID, gr.Permission); err != nil {
+					return sdk.WrapError(err, "importGroupsInApplication> Cannot insert group %s in this application %s", gr.Group.Name, app.Name)
+				}
+			}
+		} else { // add new group
+			for _, gr := range groupsToAdd {
+				groupsInApp, errLoad := group.LoadGroupsByApplication(tx, app.ID)
+				if errLoad != nil {
+					return sdk.WrapError(errLoad, "importGroupsInApplication> Cannot load groups in application %s", app.Name)
+				}
+
+				_, errGr := group.GetIdByNameInList(groupsInApp, gr.Group.Name)
+				if errGr == nil {
```

Comment:
!= nil

Reply:
No, it's normal, it doesn't return an error if the group exist in the list, so in this case the group exist and without force flag I don't replace the existing group